### PR TITLE
Update documents with current information.

### LINKED ISSUE #86: Update references to LGTM
# 🐛 Summary #
LGTM was a code-scanning tool that no-longer exists under that name. It is now built into GitHub Actions under the name, CodeQL. References to LGTM are in two files: /README.md and /languages/python/README.md.

## Proposed Solution ##
1. Implement CodeQL scanning in all cisagov repositories (if not already done).
2. Update README files to replace references to LGTM with CodeQL.

## Source ##
[The next step for LGTM.com: GitHub code scanning!](https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/)
[About code scanning with CodeQL](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql)

### PR DESCRIPTION
**Pull request by jmorrowomni**
- CONTRIBUTING.md: Added section regarding QA and code review
- README.md: Updated LGTM with CodeQL information
- dev_envs/mac-env-setup.md: Noted code is out of date
- languages/python/README.md: Updated LGTM with CodeQL information
- onboarding-checklist.md: Updated Training Site information

#  #

## 🗣 Description ##

Review the Development Guide and update where appropriate.  The two main focuses are:

- Addition of Quality Assurance (QA) to the code contribution workflow
- New hire information and configuration

## 💭 Motivation and context ##

This documentation has not been updated recently, and as a new hire following this guide
and the first member of the QA team, there were updates and insights I was able to provide
to improve this guide for future users.

Resolves #86.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

### CODE REVIEW

**Comment by jsf9k** [comment]
Please fill out the PR template.  Also remove any checkbox lines that do not apply.

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]: Remember that code contributors may be third parties that are creating whole new projects, or they may be folks who are fixing a typo in a README.  The latter probably do not need to engage prior to any code submissions, although their PRs will certainly be subject to all the QA jazz.

Regarding the paragraph that begins, "Third parties are responsible...", I don't understand what is meant by "...coordinating initial code reviews between
members of their support teams...".  Can you make that a little clearer?

Finally, I don't think there needs to be a carriage return between "...will go through out quality assurance team." and "Code contributors are welcome..."  This document should be able to be read clearly both as plain text and when rendered as Markdown.

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]:
```suggestion
submitted code will go through our quality assurance (QA) team.
```

File: README.md
Comment (jsf9k) [inline README.md]: I think this can be removed.
```suggestion
```

File: languages/python/README.md
Comment (jsf9k) [inline languages/python/README.md]:
```suggestion
[CodeQL](https://codeql.github.com/) and [Snyk](https://snyk.io).
```

File: onboarding-checklist.md
Comment (jsf9k) [inline onboarding-checklist.md]:
```suggestion
- PALMS has been retired, and is in the process of being replaced by the
```

File: onboarding-checklist.md
Comment (jsf9k) [inline onboarding-checklist.md]:
```suggestion
*Note that the issuance of Trio accounts is currently on hold.*
```

File: dev_envs/mac-env-setup.md
Comment (jsf9k) [inline dev_envs/mac-env-setup.md]: "This code" is too vague here.
```suggestion
NOTE (DEC 2022):  The [CISA `laptop` script](https://github.com/cisagov/laptop/) is out of date but still functional.
```

File: README.md
Comment (jsf9k) [inline README.md]:
```suggestion
  to automate security checks and help prevent critical vulnerabilities.  Replaces [LGTM](https://lgtm.com).
```

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]: [We only capitalize the first word of secondary headings](https://github.com/cisagov/development-guide/blob/develop/style-guide.md#headers-and-headings):
```suggestion
## Quality assurance/code reviews ##
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I noticed several minor things after doing a quick pass.

File: CONTRIBUTING.md
Comment (jsf9k) [reply CONTRIBUTING.md]: Also, I think there is some conflation between "code contributors" and "third parties" here.  We need to either use the same nomenclature everywhere or make the definitions clear.

**Comment by jmorrowomni** [comment]
> Please fill out the PR template. Also remove any checkbox lines that do not apply.

Updated the checklist.  I can't seem to add labels though, so that one is still unchecked.

File: README.md
Comment (jmorrowomni) [reply README.md]: I was going back and forth between removing the references now or waiting until we were completely done.  I'm fine either way, so will defer to your recommendation.

File: onboarding-checklist.md
Comment (jmorrowomni) [reply onboarding-checklist.md]: @jsf9k The current site is an interim replacement until a new final solution is in place.  I can reword it, but don't think your recommendation is accurate

Comment (jsf9k) [reply onboarding-checklist.md]: I see.  Maybe change "currently" to "temporarily" then?

**Comment by jsf9k** [comment]
You're going to have to fix the broken linters at some point.  Or I can do it, which I guess I will do now.

**Comment by jsf9k** [comment]
You still need to actually fill out the PR template, instead of just dumping a bulleted list above it.

File: onboarding-checklist.md
Comment (jsf9k) [reply onboarding-checklist.md]: Fixed in commit d434617.

**Comment by jsf9k** [comment]
Also, PR titles should be in the imperative mood, e.g. "Update documents with current information".

**Comment by jmorrowomni** [comment]
@jsf9k Title of the PR has been fixed, and PR template actually filled out this time

File: onboarding-checklist.md
Comment (dav3r) [inline onboarding-checklist.md]: Add period at end to match other list items:
```suggestion
[CISA Course Management Site](https://usdhs.sharepoint.com/sites/CISA_OCLO_Intranet/CISACourses/SitePages/LearningTeamHome.aspx).
```

**Review by dav3r — APPROVED** [review]
Approved (assuming @jsf9k's concerns are addressed), though I noted one tiny thang.

File: CONTRIBUTING.md
Comment (jsf9k) [reply CONTRIBUTING.md]: This conversation still hasn't been addressed.  First the language needs to be made clearer by @jmorrowomni, since I don't fully understand what is being said, and after that @cisagov/team-ois needs to acknowledge that it agrees with the content.

Comment (dav3r) [reply CONTRIBUTING.md]: @jmorrowomni - have you had a chance to clarify the wording here?  Do you have any questions or blockers that we can assist you with?

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]: Grammatical things and minor language changes to make it read more smoothly.
```suggestion
## Quality assurance and code reviews ##

In order to maintain standardization of practices, ensure security standards
are being met, and incorporate third party code as seamlessly as possible,
submitted code will go through our quality assurance (QA) team.

Code contributors are able to coordinate with the quality assurance team
at any point during the contribution process. We recommend initiating the discussions
as early as possible, to decrease the likelihood of issues around merging or
using the contributed code occurring late in the process. However, the QA team is not
responsible for the success of the project or for ensuring that all team members follow
the development standards which have been established. Any discussions or
initial input are a courtesy evaluation, and contributing teams remain
responsible for the quality of their code, internal coordination,
and alignment with the standards set forth in this guide.
```

I still feel like this section addresses large scale contributors ("I'm want to create a new `cisagov` repo") but does not address/apply to small scale contributors ("I want to fix a typo in the documentation in an existing `cisagov` repo" or "I want to add an additional test to an existing `cisagov` repo").

File: CONTRIBUTING.md
Comment (jsf9k) [inline CONTRIBUTING.md]: Grammar and language tweaks:
```suggestion
## Quality assurance and code reviews ##

In order to maintain standardization of practices, ensure security standards
are being met, and to incorporate third party code as seamlessly as possible,
all submitted code will go through our quality assurance (QA) team.

Code contributors are able to coordinate with the QA team at any point during
the contribution process. We recommend initiating the discussions as early as
possible, to decrease the likelihood of issues around merging or using the
contributed code occurring late in the process. However, the QA team is not
responsible for the success of the project or for ensuring that all team
members follow the development standards which have been established. Any
discussions or initial inputs are a courtesy evaluation, and contributing teams
remain responsible for the quality of their code, internal coordination, and
alignment with the standards set forth in this guide.

The type of contribution being made (e.g. typo corrections vs a new repository),
complexity of code change (e.g. adding a new test vs adding a new function), and
the testability of the code (e.g. well-documented and replicable) will factor
into the level of interaction needed with the QA team.
```

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (jsf9k) [reply CONTRIBUTING.md]: The new wording does address the scale of the contribution, but it does not take into account the grammatical jazz I suggested [here](https://github.com/cisagov/development-guide/pull/79#discussion_r1089397533).  I have repeated those suggestions (with a few additions) [here](https://github.com/cisagov/development-guide/pull/79#discussion_r1129741363).

File: CONTRIBUTING.md
Comment (jsf9k) [reply CONTRIBUTING.md]: Resolved in commit f34b45e.

File: CONTRIBUTING.md
Comment (jsf9k) [reply CONTRIBUTING.md]: Resolved in commit f34b45e.

**Review by jsf9k — APPROVED** [review]
Approved, but given the nature of some of the changes I'd like to get positive reviews from @mcdonnnj and @felddy before merging.

**Review by dav3r — APPROVED** [review]
Still LGTM 👍

File: onboarding-checklist.md
Comment (felddy) [inline onboarding-checklist.md]: PALMS is dead.  I think you can just remove this item entirely.

The new "hotness" is here: [CISA Course Management Site](https://usdhs.sharepoint.com/sites/CISA_OCLO_Intranet/CISACourses)

I think the specific training is here: [Cybersecurity Awareness Training ](https://usdhs.sharepoint.com/sites/CISA_OCLO_Intranet/CISACourses/SitePages/DHS-Mandatory-Training---CSAT-FY22.aspx)

_Note: as soon as you document these URLs they will be rotated to new random values._ 

Feel free to resolve this comment In absentia.

**Review by felddy — APPROVED** [review]
Looks fine to me!  
💪

File: dev_envs/mac-env-setup.md
Comment (mcdonnnj) [inline dev_envs/mac-env-setup.md]: I'd like to see the link to `cisagov/laptop` be turned into a reference-style link since it is used in multiple places.

File: languages/python/README.md
Comment (mcdonnnj) [inline languages/python/README.md]: Should we mention dependabot here?

File: onboarding-checklist.md
Comment (jsf9k) [reply onboarding-checklist.md]: Please see commit fe41c6e.

File: dev_envs/mac-env-setup.md
Comment (jsf9k) [reply dev_envs/mac-env-setup.md]: Please see commit 41e2fa9.

File: languages/python/README.md
Comment (jsf9k) [reply languages/python/README.md]: Please see commit ca3417d.

File: README.md
Comment (mcdonnnj) [inline README.md:38]: We should add [Dependabot](https://docs.github.com/en/code-security/dependabot) to this list while we're updating things.

File: README.md
Comment (mcdonnnj) [inline README.md:27]: Please move this list item so that the list is alphabetized once again.

File: languages/python/README.md
Comment (mcdonnnj) [inline languages/python/README.md]: I think this URL is more helpful than the GitHub org that houses the code.

```suggestion
[Dependabot](https://docs.github.com/en/code-security/dependabot).  By **default** all
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: README.md
Comment (jsf9k) [reply README.md:27]: Please see commit 6cff38b.

File: README.md
Comment (jsf9k) [reply README.md:38]: Please see commit 708f29c.

File: CONTRIBUTING.md
Comment (mcdonnnj) [inline CONTRIBUTING.md]: Per the conversation in https://github.com/cisagov/development-guide/pull/87#discussion_r1164360732:

```suggestion
The type of contribution being made (e.g. typo corrections vs. a new repository),
complexity of code change (e.g. adding a new test vs. adding a new function), and
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: languages/python/README.md
Comment (mcdonnnj) [inline languages/python/README.md]: This is not a blocking thought but do we want to sort these alphabetically?

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔ One comment for your consideration, but its resolution does not block my approval.

File: languages/python/README.md
Comment (jsf9k) [reply languages/python/README.md]: Please see commit f9d28c5.

**Pull request merged**
